src/runtime_src/core/include/xrt/xrt_aie.h

```diff
@@ -163,6 +163,58 @@ class bo : public xrt::bo
   }
 };
 
+class event_impl;
+class event 
+{
+public:
+
+  /**
+   * event() - Constructor from a device
+   *
+   * @param device
+   *  The device on which the profiling event should start
```

Comment:
This is a bit too obfuscated.   Line 175 is the first that hints at this having something to do with profiling.
The class name `event` is rather generic,  but I think this event is specifically for profiling.  Would it be better to make that clear by moving event under `namespace xrt::aie::profiling` ?    And if so then change `start/read/stop_profiling` to just `start/read/stop` ?

Sorry for being picky here, but this is a public API that we can't change in the future, so we better get it right the first time.

What happens when you create an event?  What is an event?

Reply:
Noted. Thank you for the suggestion Soren. Also for clarity on the event APIs, I'm referring [UG-1076](https://docs.xilinx.com/r/en-US/ug1076-ai-engine-environment/Event-Profile-APIs-for-Graph-Inputs-and-Outputs)